Enumerate differences with Shelley (#993)
IntersectMBO/formal-ledger-specifications: build-tools/static/md/mkdocs/docs/EarlierEras.md

---
source_path: build-tools/static/md/common/src/EarlierEras.md
---

## Earlier Eras

> Many of the material in the Conway specification is brought from the spefication of

Reviewer: ```suggestion
Many of the material in the Conway specification is brought from the specification of
```

earlier eras. In this section we provide an account of how the earlier specifications

> have been modified.

Reviewer: I don't think is clear what is meant by "have been modified". Earlier specifications haven't been modified, but that some things from them have been refined/adapted/improved in subsequent eras (and thus in Conway they appear possibly modified)

#### `NEWEPOCH`{.AgdaDatatype} (Figure 57)

The proof of the net neutrality of the reward update has been moved to the
`RewardUpdate`{.AgdaDatatype}.

The `PoolDistr`{.AgdaDatatype} data type has been replaced with the richer
`PoolDelegatedStake`{.AgdaDatatype} data type.

The use of the `MIR` rule has been removed as it is phased out by Conway. (?)

#### `EPOCH`{.AgdaDatatype} (Figure 45)

The use of the `NEWPP`{.AgdaDatatype} rule has been removed, as it was phased
out by Conway. (?)

#### `LEDGER`{.AgdaDatatype} (Figure 31)

Shelley assumes the transaction is valid

A use of rule `DELEGS`{.AgdaDatatype} is added in Conway.

A use of rule `CERTS`{.AgdaDatatype} is added in Conway.

#### `UTXOW`{.AgdaDatatype} (Figure 20)

The quorum check has been removed by Conway. (?)

The checks about required scripts now accounts for reference scripts.

#### `UTXO`{.AgdaDatatype} (Figure 16)

The use of the `PPUP`{.AgdaDataype} rule has been removed in Conway. (?)

## References {#references .unnumbered}

**\[CVG19\]** <span id="shelley-ledger-spec"
label="shelley-ledger-spec"></span> Jared Corduan and Polina Vinogradova
and Matthias Güdemann. *A Formal Specification of the Cardano Ledger*.

> 2019.

Author: How can I reference the Babbage spec in a similar way?

---
source_path: build-tools/static/md/common/src/EarlierEras.md
---

## Earlier Eras

Many of the material in the Conway specification is brought from the spefication of
earlier eras. In this section we provide an account of how the earlier specifications
have been modified.

### Changes to Shelley

We describe changes to the specification [CVG19](#shelley-ledger-spec) by rule type.

#### `CHAIN`{.AgdaDatatype} (Figure 75)

The uses of rules `PRTCL`{.AgdaDatatype}, `TICKN`{.AgdaDatatype}, and functions
`chainChecks`{.AgdaFunction} and `prtlSeqChecks`{.AgdaFunction} have been removed

> and are handled now by the consensus specification [[link]][consensus-spec].

Reviewer 2: ```suggestion
`chainChecks`{.AgdaFunction} and `prtlSeqChecks`{.AgdaFunction} were removed from the ledger specification
and added to the consensus specification [[link]][consensus-spec].
```

#### `BBODY`{.AgdaDatatype} (Figure 72)

> The block body hash function `bbodyHash`{.AgdaFunction} has become a field of the

Reviewer 2: (I think simpler language is preferable; besides, this is nearly equivalent, and will be more accurate if it changes again.  Same comment applies to some of the suggestions below.)

`Block`{.AgdaDatatype} type.

The block header hash function `bhash`{.AgdaFunction} has become a field of the
`BHeader`{.AgdaDatatype} type.

> The block body size function `bbodySize`{.AgdaFunction} has become a field of the

Reviewer 2: ```suggestion
The block body size function `bbodySize`{.AgdaFunction} became a field of the
```

`Block`{.AgdaDatatype} type.

The block header size function `hBbsize`{.AgdaFunction} has become a field of the
`BHeader`{.AgdaDatatype} type.

The function `isOverlaySlot`{.AgdaFunction} has been removed in the Babbage

> specification [[link]][babbage-spec].

Reviewer 2: ```suggestion
The function `isOverlaySlot`{.AgdaFunction} was removed in the Babbage
era [[link]][babbage-spec].
```

#### `TICK`{.AgdaDatatype} (Figure 64)

The function `adoptGenesisDelegs`{.AgdaFunction} has been removed.

#### `NEWEPOCH`{.AgdaDatatype} (Figure 57)

The proof of the net neutrality of the reward update has been moved to the
`RewardUpdate`{.AgdaDatatype}.

> The `PoolDistr`{.AgdaDatatype} data type has been replaced with the richer

Reviewer 2: ```suggestion
The `PoolDistr`{.AgdaDatatype} data type was replaced with the richer
```

`PoolDelegatedStake`{.AgdaDatatype} data type.

> The use of the `MIR` rule has been removed as it is phased out by Conway.

Reviewer 2: ```suggestion
The `MIR` rule was phased out in the Conway era.
```

#### `EPOCH`{.AgdaDatatype} (Figure 45)

The use of the `NEWPP`{.AgdaDatatype} rule has been removed, as it was phased
out by Conway.

#### `LEDGER`{.AgdaDatatype} (Figure 31)

Shelley assumes the transaction is valid

A use of rule `DELEGS`{.AgdaDatatype} is added in Conway.

> A use of rule `CERTS`{.AgdaDatatype} is added in Conway.

Reviewer 2: ```suggestion
A use of rule `CERTS`{.AgdaDatatype} was added in Conway.
```

#### `UTXOW`{.AgdaDatatype} (Figure 20)

The quorum check has been removed by Conway, superseded by
the new governance mechanism [[link]][conway-cip].

The checks about required scripts now accounts for reference scripts.

#### `UTXO`{.AgdaDatatype} (Figure 16)

> The use of the `PPUP`{.AgdaDataype} rule has been removed in Conway, superseded by

Reviewer 2: ```suggestion
The use of the `PPUP`{.AgdaDataype} rule was removed in Conway, superseded by
```
